### DISCUSSION TITLE
How does DT identify components affected by a vulnerability?

### DISCUSSION THREAD

**Discussion by flemminglau** [DependencyTrack/dependency-track#1909] [Q&A] [answered]
We have examples where two CVSs have the same pattern but DT conclude very differently for a specific component.

Like for instance: 

We have a product using the keycloak-core:17.0.0 library.

We get from DT that it is affected by [CVE-2021-1466](https://nvd.nist.gov/vuln/detail/CVE-2022-1466) which has a CPE pattern of: `cpe:2.3:a:redhat:keycloak:*:*:*:*:*:*:*:*`
We do not get from DT that it is affected by the [CVE-2022-1245](https://nvd.nist.gov/vuln/detail/CVE-2022-1245) which has a CPE pattern of:   `cpe:2.3:a:redhat:keycloak:*:*:*:*:*:*:*:*`

I.e.  the patterns are 100% identical and both CVEs cover version  17.0.0. One affects the library and one doesn't.

What other information is DT using to determine if our application is affected or not?

**Comment by nscuro** [comment, marked as answer]
Does the component of your `keycloak-core` library have a CPE associated with it? 

If not, DT can't match the library directly against the NVD data and instead relies on services like OSS Index to do it.

**Comment by flemminglau** [comment]
I see.

And looking up the library and version in OSS Index gives me exactly that result.

I assume (realizing now) that the this is what the "Analyzer" column tells me. (That the data was found in "OSS Index")?

So I guess my question should really be directed to the OSS Index guys. Why have they defined these different when they both seem to be targeting the same.
